src/compare.py

```diff
@@ -126,6 +138,27 @@ def compare_phash(source: MatLike, capture: MatLike, mask: MatLike | None = None
     return 1 - (hash_diff / 64.0)
 
 
+def extract_and_compare_text(capture: MatLike, text: str):
+    """
+    Compares the extracted text of the given image and returns the similarity between the two texts.
+
+    @param capture: Image of any given shape as a numpy array
+    @param text: a string to match for
+    @return: The similarity between the text in the image and the text supplied as a number 0 to 1.
+    """
+    # if the string is found 1:1 in the string extracted from the image a 1 is returned.
+    # otherwise the levenshtein ratio is calculated between the two strings and gets returned.
```

Comment:
Could you see any reason to use something else than the levenshtein ratio to get a match% ?

Reply:
It was one of the first that came to my mind and it gave good comparison results (at least in my tests). So I would stick to it for now at least.

Reply:
I'm just taking a moment to think over what's the most common source of false-negatives in text recognition, especially in the context of videogames, and which distance algorithm best fits that.
I think it's a sensible algorithm as well, in case OCR splits or merges characters accidentally.
(otherwise a dead simple Hamming distance would've worked too, but the levenshtein algorithm clearly isn't the bottleneck anyway :P )